i8086.cxx

```diff
@@ -1731,17 +1737,18 @@ uint64_t i8086::emulate( uint64_t maxcycles )
             {
                 set_al( ( al() + ( ah() * _b1 ) ) & 0xff );
                 set_ah( 0 );
+                set_PSZ8( al() );
                 _bc++;
                 break;
             }
             case 0xd6: { set_al( fCarry ? 0xff : 0 ); break; } // salc (undocumented. IP protection scheme?)
             case 0xd7: // xlat
             {
-                uint8_t * ptable = flat_address8( get_seg_value(), bx );
-                set_al( ptable[ al() ] );
+                uint8_t * ptable = flat_address8( get_seg_value(), bx + al() );
+                set_al( *ptable );

```

Comment:
This only affects wraparound situations, but the end result is the same for accesses outside of the boundary.